lib/CloudinaryPlugin.js

```diff
@@ -0,0 +1,333 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2019 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const Cloudinary = require('cloudinary').v2;
+
+class CloudinaryPlugin {
+
+  constructor() {
+    this.context = null;
+  }
+
+  init(config, context) {
+    this.context = context;
+
+    this.controllers = {
+      assetsController: {
```

Comment:
can be discussed but I would remove the suffix from the controllers, which make it quite redundant for protocols like WebSocket:
`{"controller": "cloudinary/assetsController", "action": "search", "expression": "foobar"}`